### PR TITLE
Nada/FEQ 2447/: daily limit modal refactor

### PR DESCRIPTION
**Pull request by nada-deriv**
https://github.com/deriv-com/p2p/assets/122768621/acb86ea3-89c6-489e-939a-9399e1c36c13

### CODE REVIEW

File: src/components/Modals/DailyLimitModal/DailyLimitModal.tsx [hunk 1]
```diff
@@ -14,84 +13,101 @@ type TDailyLimitModalProps = {
...
-            {getModalContent()}
+            {!isLoading && (
+                <Modal.Header hideBorder hideCloseIcon>
+                    <Text color='prominent' size={headerTextSize} weight='bold'>
+                        {getModalContent().header}
+                    </Text>
+                </Modal.Header>
+            )}
+            <Modal.Body className={clsx('py-[0.8rem] px-[2.4rem]', { 'mx-auto py-[2.4rem]': isLoading })}>
+                <Text size={textSize}>{getModalContent().body}</Text>
```
Comment (hasan-deriv) [inline src/components/Modals/DailyLimitModal/DailyLimitModal.tsx:104]: Why are  we using content like this? Can't we just use reusable component?

File: src/components/Modals/DailyLimitModal/DailyLimitModal.scss [hunk 2]
```diff
@@ -1,33 +1,3 @@
...
-            margin: 1.6rem 0;
-        }
-    }
-
-    &__footer {
-        display: flex;
-        justify-content: flex-end;
-        gap: 1rem;
-    }
+    @include default-modal;
```
Comment (hasan-deriv) [inline src/components/Modals/DailyLimitModal/DailyLimitModal.scss:2]: It is recommended not to use @include in this case. But, Using @include will duplicate the CSS rules every time it is called, which can lead to larger CSS files and potential performance issues. Instead, consider sharing class names or using @extend. @extend will merge the styles without duplicating them, making the CSS more efficient and easier to maintain.


Screenshot 2024-07-04 at 11 10 18 AM

Comment (hasan-deriv) [reply src/components/Modals/DailyLimitModal/DailyLimitModal.scss:2]: @extend:

Screenshot 2024-07-04 at 11 15 45 AM

**Review by hasan-deriv — CHANGES_REQUESTED** [review]

**Review by hasan-deriv — APPROVED** [review]

File: src/components/Modals/DailyLimitModal/DailyLimitModal.scss (hunk 2 above)
Comment (nada-deriv) [reply src/components/Modals/DailyLimitModal/DailyLimitModal.scss:2]: this change would require the mixin to be changed to placeholder and would affect all the other modals. can take it up separately. 👍

File: src/components/Modals/DailyLimitModal/DailyLimitModal.tsx (hunk 1 above)
Comment (nada-deriv) [reply src/components/Modals/DailyLimitModal/DailyLimitModal.tsx:104]: this would require repetition of modal header and also text component.

**Review by ameerul-deriv — APPROVED** [review]

**Pull request merged**